### PR TITLE
Adds error if target db has the same name as current db #13099

### PR DESCRIPTION
**Pull request by SaturnTeam**
Adds error if target db has the same name as current db name to prevent data corrupt #13099

### CODE REVIEW

File: db_operations.php [hunk 1]
```diff
@@ -48,6 +48,8 @@
 
     if (! isset($_REQUEST['newname']) || strlen($_REQUEST['newname']) === 0) {
         $message = PMA\libraries\Message::error(__('The database name is empty!'));
+    } else if($_REQUEST['newname'] === $_REQUEST['db']) {
+        $message = PMA\libraries\Message::error(__('The new database name is equal to the current one!'));
```
Comment (ibennetch) [inline db_operations.php]: I suggest changing this message to "Cannot copy database to the same name. Change the name and try again."

File: templates/database/designer/database_tables.phtml [hunk 2]
```diff
@@ -27,7 +27,7 @@
                 <?php endif; ?>
                 <td class="small_tab"
                     title="<?= __('Show/hide columns'); ?>"
-                    id="id_hide_tbody_<?= $t_n_url; ?>"
+                    id="id_hide_tbody_<?= htmlspecialchars($t_n_url) ?>"
```
Comment (ibennetch) [inline templates/database/designer/database_tables.phtml]: You should really split unrelated changes in to different pull requests, even though both of these changes are only one or two lines each.

I have a database named `p&?&` and this doesn't seem to fix the problem for that or with the database name in the original error report, either.

**Review by ibennetch — CHANGES_REQUESTED** [review]

**Comment by ibennetch** [comment]
Also note that there is #13103 which is working towards changing the defaults for the the copy/rename fields.

**Review by nijel — CHANGES_REQUESTED** [review]
As written by Isaac - try to submit one pull request with one fix. That way it's easier to merge parts of it when they are ready to get merged...

File: templates/database/designer/database_tables.phtml (hunk 2 above)
Comment (nijel) [inline templates/database/designer/database_tables.phtml]: I've cherry picked this as 4119b5ca3d2af76543ccaa5ebe1a408d2cc4fb31.

**Comment by nijel** [comment]
I've committed dbdd7bcb9db9c7a54221aec6344c8c889768aea6 based on your fix and @ibennetch wording.

**Pull request closed without merging**